packages/aws-cdk-lib/aws-route53/README.md

```diff
@@ -220,6 +220,42 @@ Constructs are available for A, AAAA, CAA, CNAME, MX, NS, SRV and TXT records.
 Use the `CaaAmazonRecord` construct to easily restrict certificate authorities
 allowed to issue certificates for a domain to Amazon only.
 
+### Health Checks
+
+Route 53 has the ability to monitor the health of your application and only return records for healthy endpoints.
+This is done using a `HealthCheck` construct.
+
+In the following example, the `ARecord` will be returned by Route 53in response to DNS queries only if the HTTP requests to the `example.com/health` endpoint return a 2XX or 3XX status code.
+
+In case, when the endpoint is not healthy, the `ARecord2` will be returned by Route 53 in response to DNS queries.
+
+```ts
+declare const myZone: route53.HostedZone;
+
+const healthCheck = new route53.HealthCheck(this, 'HealthCheck', {
+  type: route53.HealthCheckType.HTTP,
+  fqdn: 'example.com',
+  port: 80,
+  resourcePath: '/health',
+  failureThreshold: 3,
+  requestInterval: Duration.seconds(30),
+});
+
+new route53.ARecord(this, 'ARecord', {
+  zone: myZone,
+  target: route53.RecordTarget.fromIpAddresses('1.2.3.4'),
+  healthCheck,
+  weight: 100,
+});
+new route53.ARecord(this, 'ARecord2', {
+  zone: myZone,
+  target: route53.RecordTarget.fromIpAddresses('5.6.7.8'),
+  weight: 0,
```

Comment:
I'm not sure using a zero weight here is the best example, it's a pretty complex behavior with a few pitfalls, see [Weight](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-values-weighted.html#rrsets-values-weighted-weight) and [How Amazon Route 53 chooses records when health checking is configured](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/health-checks-how-route-53-chooses-records.html)

```suggestion
  weight: 50,
```